### ISSUE TITLE
Cannot return a tuple

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#337]
This should be possible if the tuple is the only object returned as the result of a function can be stored in a tuple

**Comment by yguclu** [comment]
Is this different in practice from a function with multiple return arguments? Can you add a short example?

**Comment by EmilyBourne** [comment]
In the case where only a tuple is returned then it shouldn't be. E.g:
```
def tuple_inhomogeneous_return():
    ai = (7.5, False, 8)
    return ai

def tuple_homogeneous_return():
    ai = (7.5, 4.2, 8)
    return ai
```

However of course if the output contains multiple objects the case is more complicated and probably shouldn't be handled by pyccel:
```
def tuple_homogeneous_return():
    ai = (7.5, 4.2, 8)
    b = 4
    return ai, b

a,b = tuple_homogeneous_return()
```

**Comment by yguclu** [comment]
If we think of a tuple as a "symbolic" object that is used to package variables, then its contents can always be unpacked and serialized, no matter how many layers of tuples we are using. So it should be no problem to have
```
def f():
    ai = (7.5, 4.2, 8)
    b = 4
    return ai, b

ci, d = f()

print(ci[0], ci[1], ci[2])
print(d)
```
We can now imagine to have an unpacked version of `f` that is easy to convert to Fortran:
```
def f_unpacked():
    ai = (7.5, 4.2, 8)
    b = 4
    return ai[0], ai[1], ai[2], b

out = f()
ci = (out[0], out[1], out[2])
d = out[3]

print(ci[0], ci[1], ci[2])
print(d)
```

**Comment by EmilyBourne** [comment]
> ```
> def f():
>     ai = (7.5, 4.2, 8)
>     b = 4
>     return ai, b
> 
> ci, d = f()
> 
> print(ci[0], ci[1], ci[2])
> print(d)
> ```

This works as long as all the code is translated. However if we epyccelize the function f then we will be unable to do `ci,d = f()`

**Comment by yguclu** [comment]
That's because the Fortran code does not contain any information about the tuple arguments, and we do not provide the missing information to f2py (assuming that f2py could do something about it, which I doubt).

**Comment by yguclu** [comment]
But we can worry about f2py later. If we support calling the function `f` from the same pyccelized module I would be happy already!

**Comment by EmilyBourne** [comment]
I agree this will not be possible with f2py. However if we plan to extend and use the method in `pyccel/codegen/cwrapper.py` for fortran as well as C then it would become possible at that point. It therefore seems reasonable to allow this within a module.
Although I am wary of supporting something with pyccel and not with epyccel. It could be confusing for users

**Comment by EmilyBourne** [comment]
Once issue #370 is fixed then this can be supported by both pyccel and epyccel

**Comment by EmilyBourne** [comment]
Blocked by #886
